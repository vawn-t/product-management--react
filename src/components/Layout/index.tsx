// Libraries
import React, { memo, ReactNode } from 'react';

// Layouts
import Header from '@layouts/Header';

export interface LayoutProps {
  children: ReactNode;
}

const Layout: React.FC<LayoutProps> = ({ children }) => (
  <>
    <Header />
    <main className='main'>{children}</main>
  </>
);

export default memo(Layout);
