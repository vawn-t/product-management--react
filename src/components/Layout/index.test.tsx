// Libraries
import React from 'react';
import { cleanup } from '@testing-library/react';
import { create } from 'react-test-renderer';

// Components
import Layout, { LayoutProps } from './index';

describe('Layout component', () => {
  const defaultProps: LayoutProps = {
    children: <p>Children</p>
  };
  let container: HTMLElement;
  beforeEach(() => {
    // setup a DOM element as a render target
    container = document.createElement('div');
    document.body.appendChild(container);
  });

  afterEach(() => {
    // cleanup on exiting
    container.remove();
    jest.clearAllMocks();
    cleanup();
  });

  test('should render correctly', () => {
    const defaultTree = create(<Layout {...defaultProps} />).toJSON();
    expect(defaultTree).toMatchSnapshot();
  });
});
