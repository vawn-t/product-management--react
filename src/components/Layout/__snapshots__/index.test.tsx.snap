// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout component should render correctly 1`] = `
Array [
  <header
    className="header"
  >
    <div
      className="container header-content"
    >
      <a
        className="logo-link"
      >
        <img
          alt="Page logo"
          className="image image-logo "
        />
      </a>
    </div>
  </header>,
  <main
    className="main"
  >
    <p>
      Children
    </p>
  </main>,
]
`;
